### PR TITLE
Development

### PR DESCRIPTION
**Pull request by ha020285mdv**
review, please

### CODE REVIEW

**Comment by ha020285mdv** [comment]
a little reminder

File: myAPI/API/resources.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+class WriterViewSet(ModelViewSet):
+    queryset = Writer.objects.all()
+    serializer_class = WriterSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        if self.request.method == 'GET':
+            book_name = self.request.query_params.get('book_name')
+        if book_name:
```
Comment (PonomaryovVladyslav) [inline myAPI/API/resources.py:16]: referenced before assignment, if request not get book_name will not exist

File: myAPI/API/resources.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    def get_serializer_class(self):
+        if self.request.method == 'GET':
+            return BookReadSerializer
+        return BookWriteSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        if self.request.method == 'GET':
+            author_age = self.request.query_params.get('author_age')
+        queryset = queryset.filter(writer__age__gte=int(author_age)) if author_age else queryset.all()
```
Comment (PonomaryovVladyslav) [inline myAPI/API/resources.py:40]: the same

**Review by PonomaryovVladyslav — CHANGES_REQUESTED** [review]

File: myAPI/API/resources.py (hunk 1 above)
Comment (ha020285mdv) [reply myAPI/API/resources.py:16]: as I found in docs, we get "query_params" dictionary with all sorts of requests, not only with GET, so deleting line:
      `if self.request.method == 'GET':`
solves the problem and makes code simpler

but if we strongly have to handle only GET - have to assign with None value above IF

**Review by PonomaryovVladyslav — APPROVED** [review]

**Pull request merged**